### PR TITLE
[SYCLomatic] Classify migration rules and enable rules only when needed

### PR DESCRIPTION
**Pull request by ziranzha**

### CODE REVIEW

File: clang/lib/DPCT/InclusionHeaders.h [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+  RK_Common = 0,
+  RK_Sparse,
+  RK_BLas,
+  RK_Solver,
+  RK_Rng,
+  RK_FFT,
+  RK_DNN,
+  RK_NCCL,
+  RK_Libcu,
+  RK_Thrust,
```
Comment (zhimingwang36) [inline clang/lib/DPCT/InclusionHeaders.h:35]: Add rule kind for: CUB, WMMA

File: clang/lib/DPCT/InclusionHeaders.h [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+  NUM
+};
+
+struct DpctInclusionInfo {
+  enum InclusionFlag {
+    HPF_MarkInserted,
```
Comment (zhimingwang36) [inline clang/lib/DPCT/InclusionHeaders.h]: Please update the prefix of enum values.

**Review by AndyCHHuang — APPROVED** [review]
LGTM

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**